Assets/GothicVR/Scripts/World/SunMover.cs

```diff
@@ -0,0 +1,61 @@
+using GVR.Demo;
+using GVR.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SunMover : MonoBehaviour
+{
+    private DebugSettings.SunMovementPerformance sunPerformanceSetting;
+    public GameObject sun;
+
+    public DateTimeGameEvent secondChangeEvent;
+    public DateTimeGameEvent minuteChangeEvent;
+    public DateTimeGameEvent hourChangeEvent;
+
+    private void OnEnable()
+    {
+
+    }
+    private void OnDisable()
+    {
+        switch (sunPerformanceSetting)
+        {
+            case DebugSettings.SunMovementPerformance.EveryIngameSecond:
+                secondChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameMinute:
+                minuteChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameHour:
+                hourChangeEvent.OnEventRaised -= RotateSun;
+                break;
+            default:
+                break;
+        }
+    }
+    private void Start()
+    {
+        sunPerformanceSetting = SingletonBehaviour<DebugSettings>.GetOrCreate().SunMovementPerformanceValue;
+        switch (sunPerformanceSetting)
+        {
+            case DebugSettings.SunMovementPerformance.EveryIngameSecond:
+                secondChangeEvent.OnEventRaised += RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameMinute:
+                minuteChangeEvent.OnEventRaised += RotateSun;
+                break;
+            case DebugSettings.SunMovementPerformance.EveryIngameHour:
+                hourChangeEvent.OnEventRaised += RotateSun;
+                break;
+            default:
```

Comment:
I recommend throwing an exception or add a debug.error statement. Otherwise we won't get informed when there is a new setting and sun movement is broken based on this new "unrecognized" setting.

Reply:
I actually don't know what to check for.
Well it doesn't look like this anymore anyways.

Reply:
The idea is:
```csharp
switch()
    case ...:
        ...
        break;
    default:
        Debug.LogError($"SunMovement setting {sunPerformanceSetting} isn't handled right now. Therefore sun won't move.");
        break;
```